media-proxy/include/mesh/st2110tx.h

```diff
@@ -0,0 +1,153 @@
+#ifndef ST2110TX_H
+#define ST2110TX_H
+
+#include "st2110.h"
+#include <algorithm>
+
+namespace mesh::connection {
+
+/**
+ * ST2110
+ *
+ * Base abstract class of ST2110. ST2110_20Tx/ST2110_22Tx/ST2110_30Tx
+ * inherit this class.
+ */
+template <typename FRAME, typename HANDLE, typename OPS> class ST2110Tx : public ST2110 {
+  public:
+    ST2110Tx()
+    {
+        _kind = Kind::transmitter;
+
+        _handle = nullptr;
+        _ops = {0};
+        _transfer_size = 0;
+    };
+    ~ST2110Tx() { shutdown(_ctx); };
+
+  protected:
+    HANDLE _handle;
+    OPS _ops;
+    uint32_t _transfer_size;
+    context::Context _ctx;
+
+    virtual FRAME *get_frame(HANDLE) = 0;
+    virtual int put_frame(HANDLE, FRAME *) = 0;
+    virtual HANDLE create_session(mtl_handle, OPS *) = 0;
+    virtual int close_session(HANDLE) = 0;
+
+    Result on_establish(context::Context &ctx) override
+    {
+        _ctx = context::WithCancel(ctx);
+        _stop = false;
+
+        _handle = create_session(_st, &_ops);
+        if (!_handle) {
+            log::error("Failed to create session");
+            set_state(ctx, State::closed);
+            return set_result(Result::error_general_failure);
+        }
+
+        set_state(ctx, State::active);
+        return set_result(Result::success);
+    };
+
+    Result on_shutdown(context::Context &ctx) override
+    {
+        _ctx.cancel();
+
+        if (_handle) {
```

Comment:
Consider renaming this to `session`.

Reply:
Renamed "_handle" to "mtl_session"
Fixed in e69185dab06ea1983303046f28572d6baf467706